app/services/clients/products_client.js

```diff
@@ -96,9 +96,29 @@ function getProductsByGatewayAccountId (gatewayAccountId) {
   }).then(products => products.map(product => new Product(product)))
 }
 
+/**
+ * @param {String} gatewayAccountId: the external id of the product you wish to retrieve
+ * @returns {Promise<Product>}
+ */
+function updateServiceNameOfProductsByGatewayAccountId (gatewayAccountId, serviceName) {
+  return baseClient.patch({
+    headers,
+    baseUrl,
+    url: `/gateway-account/${gatewayAccountId}`,
+    json: true,
+    body: {
+      op: 'update',
+      path: 'service_name',
+      value: serviceName
+    },
+    description: `update a product's service name`,
+    service: SERVICE_NAME
+  }).then(product => new Product(product))
```

Comment:
You won't get a product back in this case. technically you are not calling product resource. It is an implicit gateway account resource you are invoking. So there is no way to return a single product.